src/components/useVote.jsx

```diff
@@ -0,0 +1,22 @@
+import { useState } from "react";
+import { patchArticleVotes } from "../utils/api";
+
+const useVote = (article_id) => {
+  const [votedStatus, setVotedStatus] = useState(false);
+  const [isSubmitting, setIsSubmitting] = useState(false);
+
+  const handleVote = () => {
+    setIsSubmitting(true);
+    setVotedStatus((currStatus) => {
+      return !currStatus;
+    });
+    patchArticleVotes(article_id, votedStatus)
+      .then(() => {
```

Comment:
Ideally we want some logic to  handle if the patch fails. If it does, we need to update the user!